views/js/pciCreator/ims/mathEntryInteraction/runtime/mathEntryInteraction.js

```diff
@@ -112,11 +116,55 @@ define([
         }
     };
 
-    var mathEntryInteractionFactory = function () {
+    function propertyManagerFactory() {
+        return {
+            list: new Map(),
+            get(index) {
+                return this.list.get(index);
+            },
+            getFirstItem(index) {
+                if (typeof index === 'undefined') {
+                    const [inputIndex] = this.list.keys();
+                    return this.list.get(inputIndex);
+                }
+                return this.list.get(index);
+            },
+            getFirstIndex(index) {
+                if (typeof index === 'undefined') {
+                    const [inputIndex] = this.list.keys();
+                    return inputIndex;
+                }
+                return index;
+            },
+            set(index, value) {
+                return this.list.set(index, value);
+            },
+            delete(index) {
+                return this.list.delete(index);
+            },
+            keys() {
+                return this.list.keys();
+            },
+            values() {
+                return this.list.values();
+            },
+            size() {
+                return this.list.size;
+            },
+            has(index) {
+                return this.list.has(index);
+            },
+            clear() {
+                return this.list.clear();
+            }
```

Comment:
**quibble:** All API from the map is delegated, and 2 methods are added. Maybe would it be better to add those 2 methods to the map instance and return it instead, wdyt?

**suggestion:** Something like
```javascript
function inputListFactory() {
    const list = new Map();
    Object.assign(list, {
        getFirstItem(index) {
            return list.get(this.getIndex(index));
        },
        getIndex(index) {
            if (typeof index === 'undefined') {
                const [inputIndex] = list.keys();
                return inputIndex;
            }
            return index;
        }
    });
    return list;
}
```